File: Core API/Team7/Models/BookingType.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Team7.Models
{
    public class BookingType
    {
        public BookingType()
        {
            this.Schedule = new HashSet<Schedule>();
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BookingTypeID { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public int Capacity { get; set; }
        [Required]
        public string Colour { get; set; }
        public virtual ICollection<Schedule> Schedule { get; set; }
    }
}

File: Ionic/src/app/models/booking-type.ts
import {Schedule} from './schedule'
export class BookingType {
  bookingTypeID?: number;
  name: string;
  description: string;
  capacity: number;
  colour: string;
  schedule?: Schedule[];
  bookingPriceHistory?: [{
    bookingPriceID?:number,
    date?:Date,
    amount:number;
  }];
}
